pkgs/stdenv/generic/setup.sh

```diff
@@ -334,36 +334,49 @@ appendToVar() {
     fi
 }
 
-# Accumulate into `flagsArray` the flags from the named variables.
+# Accumulate flags from the named variables $2+ into the indexed array $1.
 #
-# If __structuredAttrs, the variables are all treated as arrays
-# and simply concatenated onto `flagsArray`.
-#
-# If not __structuredAttrs, then:
-#   * Each variable is treated as a string, and split on whitespace;
-#   * except variables whose names end in "Array", which are treated
-#     as arrays.
-_accumFlagsArray() {
-    local name
-    if [ -n "$__structuredAttrs" ]; then
-        for name in "$@"; do
-            local -n nameref="$name"
-            flagsArray+=( ${nameref+"${nameref[@]}"} )
-        done
-    else
-        for name in "$@"; do
+# Arrays are simply concatenated, strings are split on whitespace.
+concatTo() {
+    local -n targetref="$1"; shift
+    local name type
+    for name in "$@"; do
+        if type=$(declare -p "$name" 2> /dev/null); then
             local -n nameref="$name"
-            case "$name" in
-                *Array)
-                    # shellcheck disable=SC2206
-                    flagsArray+=( ${nameref+"${nameref[@]}"} ) ;;
+            case "${type#* }" in
+                # Deals with associative arrays by only taking the values,
+                # because that's the most sensible thing to do. Letting it
```

Comment:
Well that's the thing: from a glance, I'd think the sensible thing to do is to merge the associative arrays, given how generic the name is. I see how that's completely out of scope for the use-case, which tempts me to just prohibit the `-A` branch (make it fail) until we've figured out where it's useful. Otherwise someone will depend on the "discard the keys" behaviour and we'll be stuck with it

Reply:
> silently be caught by the subshell anyway.

Even with `set -e`? Exits for me in the interactive shell